test/bigtest/tests/circulation-rules-test.js

```diff
@@ -562,7 +566,102 @@ describe('CirculationRules', () => {
           });
 
           it('the library code should be added to the editor value', () => {
-            expect(circulationRules.editor.value).to.be.equal(`c ${libraries[0].code} `);
+            expect(circulationRules.editor.value).to.be.equal(`c ${replacer(libraries[0].code)} `);
+          });
+        });
+      });
+    });
+  });
+
+  describe('choosing criteria locating criteria - location', () => {
+    const rulesEditor = circulationRules.editor;
+    const editorHints = rulesEditor.hints;
+
+    beforeEach(async function () {
+      await rulesEditor.setValue('s ');
+    });
+
+    it('should display hints with list to select location codes', () => {
+      expect(editorHints.arePresent).to.be.true;
+    });
+
+    it('should display hints with correct header', () => {
+      expect(editorHints.header.isPresent).to.be.true;
+      expect(editorHints.header.text).to.be.equal('Select location');
+    });
+
+    it('should display three hints sections', () => {
+      expect(editorHints.sectionsCount).to.be.equals(4);
+    });
+
+    it('should display hints with correct subheaders', () => {
+      const subheaderText = sectionNumber => editorHints.sections(sectionNumber).subheader.text;
+      expect(subheaderText(0)).to.be.equal('Institution');
+      expect(subheaderText(1)).to.be.equal('Campus');
+      expect(subheaderText(2)).to.be.equal('Library');
+      expect(subheaderText(3)).to.be.equal('Location');
+    });
+
+    it('the first section should be filled by default', () => {
+      expect(editorHints.sections(0).items().length).to.be.equal(institutionsAmount);
+    });
+
+    it('other sections should be empty by default', () => {
+      expect(editorHints.sections(1).items().length).to.be.equal(0);
```

Comment:
```suggestion
      expect(editorHints.sections(1).items().length).to.equal(0);
```

It can be slightly shorter. Please handle all similar case in the file.